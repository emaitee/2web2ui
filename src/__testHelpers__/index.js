export { default as delay } from './delay';
export { default as mockStore, createMockStore } from './mockStore';
export { default as renderRowData } from './renderRowData';
