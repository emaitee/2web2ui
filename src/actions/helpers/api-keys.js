import _ from 'lodash';
import { getGrants } from 'src/selectors/api-keys';

export function formatKeyForRequest(key, getState) {
  const request = { data: {}};

  if (key.subaccount) {
    request.headers = { 'X-MSYS-SUBACCOUNT': key.subaccount.id };
  }

  request.data.label = key.label;

  if (key.grantsRadio === 'all') {
    request.data.grants = _.keys(getGrants(getState()));
  } else {
    request.data.grants = _.keys(_.pickBy(key.grants));
  }

  if (key.validIps) {
    request.data.valid_ips = key.validIps.split(',').map(_.trim);
  }

  return request;
}
