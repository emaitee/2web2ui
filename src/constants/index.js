export const DEFAULT_REDIRECT_ROUTE = '/landing';
