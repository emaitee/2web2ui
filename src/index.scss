@import '~@sparkpost/matchbox/styles.scss'; // Global Compiled
@import '~@sparkpost/matchbox/src/styles/config.scss';

// All components use REMs for sizing
// Adjusting the root font size here will scale the UI proportionally

html {
  background: color(gray, 9);
  font-size: rem(16);
  @media screen and (min-width: breakpoint(medium)) { font-size: rem(18); }
  @media screen and (min-width: breakpoint(larger)) { font-size: rem(20); }
}

::-webkit-input-placeholder { /* Chrome/Opera/Safari */
  color: #aaa;
}
::-moz-placeholder { /* Firefox 19+ */
  color: #aaa;
}
:-ms-input-placeholder { /* IE 10+ */
  color: #aaa;
}
:-moz-placeholder { /* Firefox 18- */
  color: #aaa;
}

// Overrides Critical CSS. Prevents FOUT.
#critical.ready {
  opacity: 0 !important;
  pointer-events: none !important;
}
