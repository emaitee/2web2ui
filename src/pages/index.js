export { default as AuthPage } from './auth/AuthPage';
export { default as billing } from './billing';
export { default as DashboardPage } from './dashboard/DashboardPage';
export { default as ProfilePage } from './profile/ProfilePage';
export { default as apiKeys } from './api-keys';
export { default as RegisterPage } from './register/RegisterPage';
export { default as reports } from './reports';
export { default as recipientLists } from './recipientLists';
export { default as sendingDomains } from './sendingDomains';
export { default as suppressions } from './suppressions';
export { default as subaccounts } from './subaccounts';
export { default as templates } from './templates';
export { default as users } from './users';
export { default as trackingDomains } from './trackingDomains';
export { default as webhooks } from './webhooks';
export { default as ipPools } from './ipPools';
export { default as SSOPage } from './sso/SSOPage';
export { default as ComingSoonPage } from './comingSoon/ComingSoonPage';
export { default as PageNotFound } from './pageNotFound/PageNotFound';
export { default as DefaultRedirect } from './defaultRedirect/DefaultRedirect';
