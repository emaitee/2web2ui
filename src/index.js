import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import config from './config';
import { unregister } from './helpers/registerServiceWorker';
import ErrorTracker from './helpers/errorTracker';
import rootReducer from './reducers';

import './critical.scss';
import './index.scss';
import App from './App';

const debug = () => (next) => (action) => {
  console.log(`=== ACTION: ${action.type} ===`); // eslint-disable-line no-console
  console.log(action); // eslint-disable-line no-console
  next(action);
};

// necessary for redux devtools in development mode only
const composeEnhancers = process.env.NODE_ENV !== 'production' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose; // eslint-disable-line no-mixed-operators
const store = createStore(
  rootReducer,
  composeEnhancers(
    applyMiddleware(thunk),
    applyMiddleware(debug),
    applyMiddleware(ErrorTracker.middleware)
  )
);

ErrorTracker.install(config, store);

render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById('root')
);
unregister(); // our bundle is currently too big to be added to SW cache, causing problems on every deploy

// Kill loading screen
document.getElementById('critical').className += ' ready';

/*  Google Analytics */
window.gtag('config', config.gaTag);

// appends tag manager script src to head
const gaScript = document.createElement('script');
gaScript.src = `https://www.googletagmanager.com/gtag/js?id=${config.gaTag}`;
gaScript.async = true;
document.head.appendChild(gaScript);
/*  Google Analytics End */

/**
 * Track unhandled promise rejects
 *
 * @param {PromiseRejectionEvent} event
 * @param {Error} event.reason
 */
window.onunhandledrejection = ({ reason }) => {
  ErrorTracker.report('onunhandledrejection', reason);
};
