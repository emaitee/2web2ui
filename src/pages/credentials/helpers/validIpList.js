import _ from 'lodash';
import { ipRegex } from 'src/helpers/regex';

export default function validIpList(value) {
  if (!value) {
    return;
  }

  const ips = value.split(',').map(_.trim);
  const invalidIps = ips.filter((ip) => !ipRegex.test(ip));

  if (invalidIps.length) {
    return 'Must be a comma separated list of valid IPs';
  }
}
