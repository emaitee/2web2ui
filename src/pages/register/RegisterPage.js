import React, { Component } from 'react';
import { connect } from 'react-redux';
import { withRouter, Redirect } from 'react-router-dom';
import qs from 'query-string';
import ErrorTracker from 'src/helpers/errorTracker';

import { SparkPost } from 'src/components';
import { Panel, UnstyledLink } from '@sparkpost/matchbox';
import PanelLoading from 'src/components/panelLoading/PanelLoading';
import RegisterUserForm from './RegisterUserForm';
import styles from './RegisterPage.module.scss';

import { registerUser, checkInviteToken } from 'src/actions/users';
import { authenticate, logout } from 'src/actions/auth';
import { DEFAULT_REDIRECT_ROUTE } from 'src/constants';

export class RegisterPage extends Component {

  onSubmit = (values) => {
    const { username, password } = values;
    return this.props.registerUser(this.props.token, values)
      .then(() => this.props.authenticate(username, password)
        .then(() => this.props.history.push(DEFAULT_REDIRECT_ROUTE))
        .catch((error) => {
          // user was created but auth failed, redirect to /auth
          this.props.history.push('/auth');
          ErrorTracker.report('sign-in', error);
        }))
      .catch((error) => ErrorTracker.report('register-user', error));
  }


  componentDidMount() {
    this.props.logout();
    this.props.checkInviteToken(this.props.token);
  }

  renderRegisterPanel() {
    const { invite, loading } = this.props;

    if (loading) {
      return <PanelLoading />;
    }

    if (invite.error) {
      return (
        <Panel.Section>
          <p>This invite has expired, please ask your account administrator to re-send your invitation</p>
        </Panel.Section>
      );
    }

    return (
      <Panel.Section>
        <p><small>{invite.from} invited you to join their SparkPost account.</small></p>
        <RegisterUserForm onSubmit={this.onSubmit} email={invite.email} />
      </Panel.Section>
    );
  }

  render() {
    const { token } = this.props;

    if (token === undefined) {
      return <Redirect to='/auth' />;
    }

    return (
      <div>
        <div className={styles.LogoWrapper}>
          <a href='https://www.sparkpost.com' title='SparkPost'>
            <SparkPost.Logo />
          </a>
        </div>

        <Panel accent title='Set Password'>
          { this.renderRegisterPanel() }
        </Panel>
        <UnstyledLink to='/auth'>Already signed up?</UnstyledLink>
      </div>
    );
  }
}

function mapStateToProps({ auth, users }, props) {
  return {
    token: qs.parse(props.location.search).token,
    invite: users.invite,
    loading: users.loading || auth.loginPending
  };
}

export default withRouter(connect(mapStateToProps, { registerUser, checkInviteToken, logout, authenticate })(RegisterPage));
