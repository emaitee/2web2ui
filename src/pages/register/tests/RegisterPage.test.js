import { shallow } from 'enzyme';
import React from 'react';
import * as ErrorTracker from 'src/helpers/errorTracker';
import { RegisterPage } from '../RegisterPage';
import { DEFAULT_REDIRECT_ROUTE } from 'src/constants';

// mocks
const checkInviteToken = jest.fn();
const registerUser = jest.fn(() => Promise.resolve());
const logout = jest.fn();
const authenticate = jest.fn(() => Promise.resolve());
const historyPush = jest.fn();
const report = jest.fn();

ErrorTracker.default = { report };

describe('Page: RegisterPage', () => {

  const props = {
    token: 'tokerino',
    invite: {
      error: null,
      from: 'jose@zamora.io',
      email: 'appteam@sparkpost.com'
    },
    loading: false,
    history: {
      push: historyPush
    },
    checkInviteToken,
    registerUser,
    logout,
    authenticate
  };

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<RegisterPage {...props} />);
  });

  test('happy path render', () => {
    expect(logout).toHaveBeenCalled();
    expect(checkInviteToken).toHaveBeenCalledWith(props.token);
    expect(wrapper).toMatchSnapshot();
  });

  test('loading', () => {
    wrapper.setProps({ loading: true });
    expect(wrapper).toMatchSnapshot();
  });

  test('no token', () => {
    wrapper.setProps({ token: undefined });
    expect(wrapper).toMatchSnapshot();
  });

  test('token expired', () => {
    const error = { message: 'n0o0o0o0' };
    wrapper.setProps({ invite: { error }});
    expect(wrapper).toMatchSnapshot();
  });

  describe('onSubmit', () => {
    const values = {
      username: 'some',
      password: 'values'
    };

    test('success', async() => {
      await wrapper.instance().onSubmit(values);
      expect(registerUser).toHaveBeenCalledWith(props.token, values);
      expect(authenticate).toHaveBeenCalledWith(values.username, values.password);
      expect(historyPush).toHaveBeenCalledWith(DEFAULT_REDIRECT_ROUTE);
    });

    test('authenticate user fail', async() => {
      const error = { message: 'n0o0o0o0' };
      authenticate.mockReturnValue(Promise.reject(error));
      await wrapper.instance().onSubmit(values);
      expect(registerUser).toHaveBeenCalledWith(props.token, values);
      expect(historyPush).toHaveBeenCalledWith('/auth');
      expect(report).toHaveBeenCalledWith('sign-in', error);
    });

    test('register user fail', async() => {
      const error = { message: 'n0o0o0o0' };
      registerUser.mockReturnValue(Promise.reject(error));
      await wrapper.instance().onSubmit(values);
      expect(registerUser).toHaveBeenCalledWith(props.token, values);
      expect(authenticate).not.toHaveBeenCalled();
      expect(historyPush).not.toHaveBeenCalled();
      expect(report).toHaveBeenCalledWith('register-user', error);
    });

  });

});
