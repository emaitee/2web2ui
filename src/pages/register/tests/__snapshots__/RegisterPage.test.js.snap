// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: RegisterPage happy path render 1`] = `
<div>
  <div>
    <a
      href="https://www.sparkpost.com"
      title="SparkPost"
    >
      <Logo />
    </a>
  </div>
  <Panel
    accent={true}
    title="Set Password"
  >
    <Panel.Section>
      <p>
        <small>
          jose@zamora.io
           invited you to join their SparkPost account.
        </small>
      </p>
      <Connect(ReduxForm)
        email="appteam@sparkpost.com"
        onSubmit={[Function]}
      />
    </Panel.Section>
  </Panel>
  <UnstyledLink
    to="/auth"
  >
    Already signed up?
  </UnstyledLink>
</div>
`;

exports[`Page: RegisterPage loading 1`] = `
<div>
  <div>
    <a
      href="https://www.sparkpost.com"
      title="SparkPost"
    >
      <Logo />
    </a>
  </div>
  <Panel
    accent={true}
    title="Set Password"
  >
    <PanelLoading
      minHeight="400px"
    />
  </Panel>
  <UnstyledLink
    to="/auth"
  >
    Already signed up?
  </UnstyledLink>
</div>
`;

exports[`Page: RegisterPage no token 1`] = `
<Redirect
  push={false}
  to="/auth"
/>
`;

exports[`Page: RegisterPage token expired 1`] = `
<div>
  <div>
    <a
      href="https://www.sparkpost.com"
      title="SparkPost"
    >
      <Logo />
    </a>
  </div>
  <Panel
    accent={true}
    title="Set Password"
  >
    <Panel.Section>
      <p>
        This invite has expired, please ask your account administrator to re-send your invitation
      </p>
    </Panel.Section>
  </Panel>
  <UnstyledLink
    to="/auth"
  >
    Already signed up?
  </UnstyledLink>
</div>
`;
